src/chatd.h

```diff
@@ -124,6 +124,17 @@ class Listener
      */
     virtual void onHistoryDone(HistSource /*source*/) {}
 
+    /**
+     * @brief Chat history have been truncated as result of reception of OP_RETENTION command.
+     * All messages that exceed retention time must be cleared, so we need to inform app
+     * to free all resources associated to these messages.
+     *
+     * @param msg Most recent message affected by retention time
+     * @param idx Index of the message
+     * @param status Status of the message
+     */
+    virtual void onRetentionHistoryTruncated(const Message &msg, const Idx &idx, const Message::Status &status) {}
```

Comment:
It looks to me more natural:
```suggestion
    virtual void onHistoryTruncatedByRetentionTime(const Message &msg, const Idx &idx, const Message::Status &status) {}
```